### ISSUE TITLE
Marquee selection(Box Selection) in our viewer

### ISSUE THREAD

**Issue by intrida-dev** [xeokit/xeokit-sdk#1254]
In **BIMViewer** example, I can see there is a **Marquee Select** feature. It seems like that whole feature is completely implemented for **BIMViewer** demo and not exposed in **xeokit-sdk**. We want to implement the same in our application. Can you please explain us the best way to implement the same in our app?

**Comment by xeolabs** [comment]
Unfortunately we can't help with the development of user's application code much outside of a commercial support agreement, however in this case the Marquee selection logic is pretty much encapsulated in this Marquee tool controller class here:

https://github.com/xeokit/xeokit-bim-viewer/blob/master/src/toolbar/MarqueeSelectionTool.js

Note that this controller class also depends on a kd-tree class, ObjectsKdTree3, defined here:

https://github.com/xeokit/xeokit-bim-viewer/blob/master/src/collision/ObjectsKdTree3.js

So, basically you create the k-d tree for the Viewer, and the k-d tree keeps itself populated with whatever objects are in the Viewer.

The Marquee tool controller then handles user drag input to build a box, which it then searches the k-d tree with to find intersecting objects, which it then sets as selected.

**Comment by intrida-dev** [comment]
We will follow this to implement 'Marquee Selection'. Thanks :-)

**Comment by xeolabs** [comment]
@intrida-dev Just an FYI that we're now adding components to the core SDK to support 2D marquee picking - will release these in the next couple of days, I'll let you know when: https://github.com/xeokit/xeokit-sdk/pull/1260

**Comment by xeolabs** [comment]
@intrida-dev See https://github.com/xeokit/xeokit-sdk/releases/tag/v2.4.2-beta-13

**Comment by intrida-dev** [comment]
This is good. Thanks
